src/darwin/system-darwin.m

```diff
@@ -245,22 +248,35 @@
 static void
 frida_collect_application_info_from_id_nsstring (NSString * identifier, FridaEnumerateApplicationsOperation * op)
 {
-  FridaHostApplicationInfo info;
+  FridaHostApplicationInfo info = { 0, };
   FridaScope scope = op->scope;
   FridaSpringboardApi * api = op->api;
   NSString * name;
   struct kinfo_proc * process;
 
   name = api->SBSCopyLocalizedApplicationNameForDisplayIdentifier (identifier);
+  if (!name)
+  {
+    LSApplicationProxy * app = [api->LSApplicationProxy applicationProxyForIdentifier:identifier];
+    name = [[app localizedNameWithPreferredLocalizations:nil useShortNameOnly:YES] retain];
+  }
 
   info.identifier = g_strdup (identifier.UTF8String);
-  info.name = g_strdup (name.UTF8String);
-  info.pid = 0;
+  info.name = g_strdup (name ? name.UTF8String : "");
   info.parameters = frida_make_parameters_dict ();
 
   process = g_hash_table_lookup (op->process_by_identifier, info.identifier);
   if (process != NULL)
+  {
     info.pid = process->kp_proc.p_pid;
+  }
+  else if (api->FBSSystemService != nil)
+  {
+    FBSSystemService * service = [api->FBSSystemService sharedService];
+    gint pid = [service pidForApplication:identifier];
```

Comment:
Can inline `[api->FBSSystemService sharedService]` here to avoid violating the no-declaration-after-statement style rule. (And not have to make this code too verbose to fix the style.)